src/ui/MD/MD.tsx

```diff
@@ -11,14 +11,40 @@ import rehypeRaw from "rehype-raw";
 import { FilePath } from "../../Paths/FilePath";
 import { getPage } from "../../Documentation/root";
 
-export function MD(props: { pageFilePath: FilePath; top: number }): React.ReactElement {
-  const pageContent = getPage(props.pageFilePath);
+export function MD({
+  pageFilePath,
+  top,
+  modalWrapperRef,
+}: {
+  pageFilePath: FilePath;
+  top: number;
+  /**
+   * If this component is used for rendering the documentation popup, this parameter is the ref of the wrapper div in
+   * the modal.
+   */
+  modalWrapperRef?: React.RefObject<HTMLDivElement>;
+}): React.ReactElement {
+  const pageContent = getPage(pageFilePath);
 
+  /**
+   * After rendering the doc page, we need to scroll it. This component is used in 2 places:
+   * - src\Documentation\ui\DocumentationRoot.tsx: The documentation tab.
```

Comment:
Can we not put the scrolling logic in these files, or pass it in via a prop or something? The "if" behavior is rather icky.